src/ecs/components/Direction.hpp

```diff
@@ -15,7 +15,9 @@ namespace ecs::component
      * y represents the y-axis (vertically)
      */
     struct Direction {
+        Direction(int x, int y, bool hasMoved = 0) : x(x), y(y), hasMoved(hasMoved) {}
         int x;
         int y;
+        bool hasMoved;
```

Comment:
Not sure of the utility of the hasMoved member, please update documentation if it's wrong @aurelenc

Reply:
Required for optimization purposes